src/patch.js

```diff
@@ -1,13 +1,21 @@
 var callbacks = []
 
-export function patch(parent, oldNode, newNode) {
-  var element = patchElement(parent, parent.children[0], oldNode, newNode)
+export function mount(parent, node) {
+  var element = parent.appendChild(createElement(node))
 
   for (var cb; (cb = callbacks.pop()); cb()) {}
 
   return element
 }
 
+export function patch(oldNode, newNode) {
+  newNode._ref = patchElement(oldNode._ref.parentNode, oldNode._ref, oldNode, newNode)
```

Comment:
@estrattonbailey Can we avoid doing this? I don't want to save stuff like this on the node. 🤔

Reply:
Ah, I meant to mention that above. The main issue I saw with patching subtrees was that there was no reference to the real DOM node for any given tree of virtual nodes, so `patchElement` didn't have any root node to mutate childnodes from.

Do you think there's another way to get a DOM reference for an arbitrary virtual node? In terms of memory consumption, maybe a `getter` would help. Or maybe you see an entirely different way of doing this?